mlir/lib/Conversion/MeshToMPI/MeshToMPI.cpp

```diff
@@ -390,8 +698,25 @@ struct ConvertUpdateHaloOp
         offsets[dim] = orgOffset;
       };
 
-      genSendRecv(false);
-      genSendRecv(true);
+      auto get_i32val = [&](OpFoldResult &v) {
+        return isa<Value>(v)
+                   ? cast<Value>(v)
+                   : rewriter.create<arith::ConstantOp>(
+                         loc,
+                         rewriter.getI32IntegerAttr(
+                             cast<IntegerAttr>(cast<Attribute>(v)).getInt()));
+      };
```

Comment:
Nit: Why not wrap the entire loop body in this lambda and just call it with 0 and 1?

Reply:
ok